src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -134,13 +137,19 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 	Assert(masterQuery->hasAggs || masterQuery->groupClause);
 
 	aggregateTargetList = masterQuery->targetList;
+	havingQual = masterQuery->havingQual;
+
 	count_agg_clauses(NULL, (Node *) aggregateTargetList, &aggregateCosts);
+	count_agg_clauses(NULL, havingQual, &aggregateCosts);
```

Comment:
Not sure if @anarazel is planning to send out a separate fix, but since we're adding code here, we should add `MemSet(&aggregateCosts, 0, sizeof(AggClauseCosts));`. Otherwise, we have uninitialised values in aggregateCosts.